packages/cli/src/commands/versions/bump.ts

```diff
@@ -68,67 +69,76 @@ export default async (cmd: Command) => {
   const versionBumps = new Map<string, PkgVersionInfo[]>();
   // Track package versions that we want to remove from yarn.lock in order to trigger a bump
   const unlocked = Array<{ name: string; range: string; target: string }>();
-  await workerThreads(16, dependencyMap.entries(), async ([name, pkgs]) => {
-    let target: string;
-    try {
-      target = await findTargetVersion(name);
-    } catch (error) {
-      if (isError(error) && error.name === 'NotFoundError') {
-        console.log(`Package info not found, ignoring package ${name}`);
-        return;
-      }
-      throw error;
-    }
 
-    for (const pkg of pkgs) {
-      if (semver.satisfies(target, pkg.range)) {
-        if (semver.minVersion(pkg.range)?.version !== target) {
-          unlocked.push({ name, range: pkg.range, target });
+  await runParallelWorkers({
+    parallelismFactor: 4,
+    items: dependencyMap.entries(),
+    async worker([name, pkgs]) {
+      let target: string;
+      try {
+        target = await findTargetVersion(name);
+      } catch (error) {
+        if (isError(error) && error.name === 'NotFoundError') {
+          console.log(`Package info not found, ignoring package ${name}`);
+          return;
         }
+        throw error;
+      }
+
+      for (const pkg of pkgs) {
+        if (semver.satisfies(target, pkg.range)) {
+          if (semver.minVersion(pkg.range)?.version !== target) {
+            unlocked.push({ name, range: pkg.range, target });
+          }
 
-        continue;
+          continue;
+        }
+        versionBumps.set(
+          pkg.name,
+          (versionBumps.get(pkg.name) ?? []).concat({
+            name,
+            location: pkg.location,
+            range: `^${target}`, // TODO(Rugvip): Option to use something else than ^?
+            target,
+          }),
+        );
       }
-      versionBumps.set(
-        pkg.name,
-        (versionBumps.get(pkg.name) ?? []).concat({
-          name,
-          location: pkg.location,
-          range: `^${target}`, // TODO(Rugvip): Option to use something else than ^?
-          target,
-        }),
-      );
-    }
+    },
   });
 
   const filter = (name: string) => minimatch(name, pattern);
 
   // Check for updates of transitive backstage dependencies
-  await workerThreads(16, lockfile.keys(), async name => {
-    // Only check @backstage packages and friends, we don't want this to do a full update of all deps
-    if (!filter(name)) {
-      return;
-    }
-
-    let target: string;
-    try {
-      target = await findTargetVersion(name);
-    } catch (error) {
-      if (isError(error) && error.name === 'NotFoundError') {
-        console.log(`Package info not found, ignoring package ${name}`);
+  await runParallelWorkers({
+    parallelismFactor: 4,
+    items: lockfile.keys(),
+    async worker(name) {
+      // Only check @backstage packages and friends, we don't want this to do a full update of all deps
+      if (!filter(name)) {
```

Comment:
feels tempting to do this filtering outside directly on the `items`, if we're only gonna actually work on a small fraction :)

Reply:
Prolly very similar perf in the end tbh, in this case we avoid creating a duplicate array since we're iterating the keys directly